lib/archethic/self_repair/sync.ex

```diff
@@ -466,16 +467,23 @@ defmodule Archethic.SelfRepair.Sync do
     Task.Supervisor.async_stream(
       TaskSupervisor,
       attestations,
-      fn attestation ->
+      fn attestation = %ReplicationAttestation{
+           transaction_summary: %TransactionSummary{address: address}
+         } ->
+        # todo: parallelize downloads
         tx = TransactionHandler.download_transaction(attestation, download_nodes)
+
+        inputs = TransactionChain.fetch_inputs(address, download_nodes)
```

Comment:
I think downloading data should be done by the TransactionHandler.
Also here you don't run the Election to fetch the inputs